qiskit/pulse/logical_elements_frames.py

```diff
@@ -0,0 +1,499 @@
+            frequency: The initial frequency set for the frame.
+            phase: The initial phase set for the frame. Default value 0.
+
+        """
+        self._frequency = frequency
+        self._phase = phase
+        super().__init__(name)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the frame."""
+        return f"GenericFrame({self._name})"
+
+    @property
+    def frequency(self) -> float:
+        """Return the initial frequency of the generic frame."""
+        return self._frequency
+
+    @property
+    def phase(self) -> float:
+        """Return the initial phase of the generic frame."""
+        return self._phase
+
+    def __eq__(self, other: "GenericFrame") -> bool:
+        """Return True iff self and other are equal, specifically, iff they have the same type
+        name, frequency and phase.
+
+        Args:
+            other: The generic frame to compare to this one.
+
+        Returns:
+            True iff equal.
+        """
+        return (
+            type(self) is type(other)
+            and self._name == other._name
+            and self._frequency == other._frequency
+            and self._phase == other._phase
+        )
+
+    def __hash__(self) -> int:
+        return self._hash
+
+
+class QubitFrame(Frame):
+    """A frame associated with the driving of a qubit.
+
+    ``QubitFrame`` is a frame associated with the driving of a specific qubit. The initial frequency of
+    the frame will be taken as the default driving frequency provided by the backend
+    during compilation.
+    """
+
+    def __init__(self, qubit_index: int):
+        """Create ``QubitFrame``.
+
+        Args:
+            qubit_index: The index of the qubit represented by the frame.
+        """
+        self._validate_index(qubit_index)
+        self._index = qubit_index
+        super().__init__(self.name)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the qubit frame."""
+        return f"QubitFrame{self.qubit_index}"
+
+    @property
+    def qubit_index(self) -> int:
+        """Return the qubit index of the qubit frame."""
+        return self._index
+
+    def _validate_index(self, index) -> None:
+        """Raise a ``PulseError`` if the qubit index is invalid. Namely, check if the index is a
+        non-negative integer.
+
+        Raises:
+            PulseError: If ``identifier`` (index) is a negative integer.
+        """
+        pass
+        if not isinstance(index, (int, np.integer)) or index < 0:
+            raise PulseError("Qubit index must be a non-negative integer")
+
+
+class MeasurementFrame(Frame):
+    """A frame associated with the measurement of a qubit.
+
+    ``MeasurementFrame`` is a frame associated with the measurement of a specific qubit.
+    If not set otherwise, the initial frequency of the frame will be taken as the default
+    measurement frequency provided by the backend during compilation.
+    """
```

Comment:
```suggestion
    """A frame associated with the dispersive readout of a qubit.

    ``MeasurementFrame`` is a frame associated with the dispersive readout of a specific qubit,
    which requires a stimulus tone driven at frequency off resonant to qubit drive.
    
    If not set otherwise, the initial frequency of the frame will be taken as the default
    measurement frequency provided by the backend during compilation.
    """
```
To avoid confusion with acquire channel.